docs/request-for-comment/2022-05-11-microservice-language-selection.md

```diff
@@ -0,0 +1,29 @@
+### Discussed in
+
+<sup>Originally posted</sup>
+- Start  2022-03-24
+- Related Issue: https://vfs.atlassian.net/wiki/spaces/IET/pages/2114191369/RFC+File+Upload+will+be+developed+in+python
+
+# Summary
+
+`vets-api` is implemented as a monolithic ruby on rails application. IET is responsible for exploring how this monolith may be broken into independent services. This presents an opportunity for teams to leverage the test best suited for the microservice. The vanguard of microservice development, IET intends to develop file upload in python. 
+
+# Motivation
+
+A core objective of a shift from monolithic architecture to microservices is to shift pieces of functionality into independently deployable, loosely coupled services. Encapsulating functionality into discrete services reduces the overall complexity of the code that each team is responsible for, which makes the overall code base more testable and maintainable. Each microservice should be capable of operating independently, which introduces the possibility of developing services in languages other than ruby on rails. As teams split services from `vets-api` they are no longer required to develop these services using ruby on rails. 
```

Comment:
There is delta for `overall code base more testable and maintainable`. For a given amount of time until microservices are stable that testing and maintenance within the monolith and the microservices in not easier. In most cases prior to stability testing is significantly more difficult. I suggest removing this statement or qualify it with a datapoint which estimates when we believe microservices will unlock your statement to be true. Certainly the first microservice will not make testing and maintenance easier however it will become easier eventually, we just need to better understand when that will be.

Reply:
This is a good point, and feeds into the question of how we want to definitively measure success of a service.  IET’s primary charter is to explore the feasibility of breaking `vets-api` and to develop strategies for the creation of microservices. I will update this document with datapoints that the platform will be able to use to determine whether or not the effort involved in breaking the monolith into microservices is worth the gains that will hypothetically made from simplifying the monolith.